packages/vue/src/components/molecules/SfModal/SfModal.js

```diff
@@ -0,0 +1,78 @@
+export default {
+  name: "SfModal",
+  props: {
+    /**
+     * Cross closing modal button
+     */
+    cross: {
+      type: Boolean,
+      default: true
+    },
+    /**
+     * Visibility of the modal
+     */
+    visible: {
+      type: Boolean,
+      default: false
+    },
+    /**
+     * Whether to show the overlay
+     */
+    overlay: {
+      type: Boolean,
+      default: true
+    },
+    /**
+     * If true clicking outside will not dismiss the modal
+     */
+    persistent: {
+      type: Boolean,
+      default: false
+    },
+    /**
+     * overlay transition effect
+     */
+    transitionOverlay: {
+      type: String,
+      default: "fade"
+    },
+    /**
+     * overlay transition effect
+     */
+    transitionModal: {
+      type: String,
+      default: "fade"
+    }
+  },
+  mounted() {
+    const keydownHandler = e => {
+      if (e.key === "Escape" || e.key === "Esc" || e.keyCode === 27) {
+        this.$emit("close");
+      }
+    };
+    document.addEventListener("keydown", keydownHandler);
```

Comment:
when the modal is closed but the instance is not destroyed (which is not necessary), you must remove this event listener, right? you're not doing that in this case...
the other option is to check inside the `keydownHandler` if the modal is visible, but it's not the best idea i guess, with multiple modals we would have multiple listeners unnecessarily

Reply:
and you should add the listener only when the modal is opened, not necessarily when the component is mounted as it can be invisible (with `visible` prop set to false)

Reply:
uh ok, I admit I copied this from sidebar component 😝

Reply:
truth, i saw it now, i think we have to fix sidebar component too

Reply:
@leomp12 on it ;)